proposals/low-level-struct-improvements.md

```diff
@@ -285,8 +294,14 @@ Span<int> Use()
 }
 ```
 
+Further treating the input to an `out` parameter as returnable is extremely confusing to developers. It essentially subverts the intent of `out` to force developers to think about the original value that the language intends to be assigned by the caller. That is why going forward languages implementing `ref` safety will be required to ensure that `out` parameters are never returned. 
+
+This means going forward `out` will match developers intuition when it comes to ref safety: `out` means `out`.
+
 The span safety rules will be written in terms of `scoped ref` and `ref`. For span safety purposes an `in` parameter is equivalent to `ref` and `out` is equivalent to `scoped ref`. Both `in` and `out` will only be specifically called out when it is important to the semantic of the rule. Otherwise they are just considered `ref` and `scoped ref` respectively.
 
+When discussing the *ref-safe-to-escape* of arguments that correspond to `in` parameters they will be generalized as `ref` arguments in the spec. In the case the argument is an lvalue then the *ref-safe-to-escape* is that of the lvalue, otherwise it is *current method*. Again `in` will only be called out here when it is important to the semantic of the current rule.
```

Comment:
>is that of the lvalue [](http://example.com/codeflow?start=214&length=21)

"is the *calling method**"?

Reply:
In the case arguments to `in` parameters are not lvalue the compiler generates a temporary and passes that by `ref`. This is passing a local by `ref` hence *ref-safe-to-escape* should be *current method*. Does that make sense or am I missing something?

Reply:
I was referring to the previous clause: 
> In the case the argument is an lvalue then the *ref-safe-to-escape* is that of the lvalue

Should that be ... ?
> In the case the argument is an lvalue then the *ref-safe-to-escape* is **the calling method**